spec/unit/puppet/type/dsc_package_spec.rb

```diff
@@ -202,12 +202,14 @@
     expect{dsc_package[:dsc_returncode] = true}.to raise_error(Puppet::ResourceError)
   end
 
-  it 'should not accept int for dsc_returncode' do
-    expect{dsc_package[:dsc_returncode] = -16}.to raise_error(Puppet::ResourceError)
+  it 'should accept int for dsc_returncode' do
+    dsc_package[:dsc_returncode] = [32, 64, 128]
+    expect(dsc_package[:dsc_returncode]).to eq([32, 64, 128])
   end
 
-  it 'should not accept uint for dsc_returncode' do
-    expect{dsc_package[:dsc_returncode] = 16}.to raise_error(Puppet::ResourceError)
+  it 'should accept uint for dsc_returncode' do
+    dsc_package[:dsc_returncode] = [32, 64, 128]
+    expect(dsc_package[:dsc_returncode]).to eq([32, 64, 128])
```

Comment:
Also keep in mind when we move over to the validation code in the helpers that there are ranges of acceptable values here (i.e. signed values have to reserve 1 bit for the sign)... so it's important to make sure tests against `int` reference `int`, etc.

Probably more cleanup to do in that subsequent PR.